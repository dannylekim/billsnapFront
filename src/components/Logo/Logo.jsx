import React, {useEffect, useState} from 'react';
import PropType from 'prop-types';
import './styles.scss';

//TODO: replace logo consts with 'src/assets/logo' images
const logoSmall = "https://via.placeholder.com/30x30.png?text=30x30+Logo";
const logoMed = "https://via.placeholder.com/60x60.png?text=60x60+Logo";
const logoLarge = "https://via.placeholder.com/100x100.png?text=100x100+Logo";

export const Logo = (prop) => {

    const [imgSrc, setImgSrc] = useState(logoSmall);

    useEffect(() => {
        switch (prop.size) {
            case 'medium':
                setImgSrc(logoMed);
                break;
            case 'large':
                setImgSrc(logoLarge);
                break;
            default:
                setImgSrc(logoSmall);
        }
    }, [prop.size]); // Empty array ensures that effect is only run on mount and unmount

    return (
        <img src={imgSrc} alt="Logo"/>
    )
};

Logo.propTypes = {
    size: PropType.string // options: small by default, 'medium', 'large'
};