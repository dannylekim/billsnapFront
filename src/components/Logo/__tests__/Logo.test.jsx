import React from 'react';
import ReactDOM from 'react-dom';
import { Logo } from '../Logo.jsx';

describe('Logo', () =>{
    describe('render', () => {
        it('renders without crashing', () => {
            const div = document.createElement('div');
            ReactDOM.render(<Logo />, div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
});