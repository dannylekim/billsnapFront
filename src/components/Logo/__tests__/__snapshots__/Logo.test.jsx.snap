// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshots ðŸ“¸ Logo should match snap shot 1`] = `
<picture>
  <source
    media="(min-width: 768px)"
    srcSet="https://via.placeholder.com/60x60.png?text=60x60+Logo"
  />
  <source
    media="(min-width: 1024px)"
    srcSet="https://via.placeholder.com/80x80.png?text=80x80+Logo"
  />
  <img
    alt="Small logo"
    src="https://via.placeholder.com/40x40.png?text=40x40+Logo"
  />
</picture>
`;
